File: python/hello/hello.py
import sys


if __name__ == "__main__":
    argv = sys.argv[1:]

    print("Hello,", end="")

    if len(argv) > 0:
        for v in argv:
            print(f' {v.capitalize()}', end="")
    else:
        print(" World", end="")

    print("!")

File: cpp/hello/hello.cpp
#include <algorithm>
#include <cctype>
#include <iostream>
#include <string>

int main(int argc, char* argv[]) {
    std::cout << "Hello,";

    if(argc > 1)
        for(int i = 1; i < argc; ++i) {
            std::string str(argv[i]);
            std::transform(str.begin(), str.begin() + 1, str.begin(),
                           [](unsigned char c) { return std::toupper(c); });
            std::cout << " " << str;
        }
    else
        std::cout << " World";

    std::cout << "!" << std::endl;

    return 0;
}
